ftw/mail/attachment.py

```diff
@@ -30,34 +27,19 @@ def render(self):
         except ValueError:
             raise NotFound
 
-        # get attachment at position pos
-        attachment = None
-        for i, part in enumerate(walk(message)):
-            if i == pos:
-                attachment = part
-                continue
-
-        if attachment is not None:
-            content_type = attachment.get_content_type()
-            filename = utils.get_filename(attachment, content_type)
-            if filename is None:
-                raise NotFound
-            # make sure we have a unicode string
-            if not isinstance(filename, unicode):
-                filename = filename.decode('utf-8', 'ignore')
-            # internet explorer and safari don't like rfc encoding of filenames
-            # and they don't like utf encoding too.
-            # therefore we first try to encode the filename in iso-8859-1
-            try:
-                filename = filename.encode('iso-8859-1')
-            except:
-                filename = filename.encode('utf-8', 'ignore')
-
-            self.request.response.setHeader('Content-Type', content_type)
-            self.request.response.setHeader('Content-Disposition', 'inline; filename=%s' % filename)
-
-            if content_type == 'message/rfc822':
-                return attachment.as_string()
-            return attachment.get_payload(decode=1)
-
-        raise NotFound
+        data, content_type, filename = get_attachment_data(self.context.msg, pos)
+
+        if filename is None:
+            raise NotFound
+
+        # internet explorer and safari don't like rfc encoding of filenames
+        # and they don't like utf encoding too.
+        # therefore we first try to encode the filename in iso-8859-1
+        try:
+            filename = filename.encode('iso-8859-1')
+        except:
```

Comment:
We know that filename is unicode (as we convert it ourselves). I would prefer to get rid of the bare try-except here and use `UnicodeEncodeError`.

Reply:
Right thanks!